fcos-policy-engine/src/utils.rs

```diff
@@ -4,24 +4,23 @@ use failure::{bail, Error, Fallible, SyncFailure};
 use reqwest::Method;
 use std::time::Duration;
 
-/// Default timeout for HTTP requests (30 minutes).
-const DEFAULT_HTTP_REQ_TIMEOUT: Duration = Duration::from_secs(30 * 60);
-/// Default address of fcos-graph-builder, which is running in the same pod.
-const DEFAULT_GB_ADDR: &str = "http://127.0.0.1:8080/v1/graph";
-
 /// Return a request builder with base URL and parameters set.
-fn new_request(method: reqwest::Method, url: reqwest::Url) -> Fallible<reqwest::RequestBuilder> {
-    let client = reqwest::ClientBuilder::new()
-        .timeout(DEFAULT_HTTP_REQ_TIMEOUT)
-        .build()?;
+fn new_request(
+    method: reqwest::Method,
+    url: reqwest::Url,
+    req_timeout: Duration,
+) -> Fallible<reqwest::RequestBuilder> {
+    let client = reqwest::ClientBuilder::new().timeout(req_timeout).build()?;
     let builder = client.request(method, url);
     Ok(builder)
 }
 
 /// Fetch the graph from the fcos-graph-builder instance with the query specified.
 pub(crate) fn fetch_graph_from_gb(
```

Comment:
`clippy` complains about this line, I tried to add `async` on this function and add an additional `.await` in https://github.com/coreos/fedora-coreos-cincinnati/pull/30/files#diff-59ebd32bdcd4e33d158ed22f4e8468c7R154, which suppress the complaint.. but I don't think this is functionally blocking anything

```
$ cargo clippy -- -D warnings
    Checking fcos-policy-engine v0.1.0 (/home/abai/git/fedora-coreos-cincinnati/fcos-policy-engine)
error: this function can be simplified using the `async fn` syntax
  --> fcos-policy-engine/src/utils.rs:19:1
   |
19 | / pub(crate) fn fetch_graph_from_gb(
20 | |     upstream_base: reqwest::Url,
21 | |     stream: String,
22 | |     basearch: String,
23 | |     req_timeout: Duration,
24 | | ) -> impl Future<Output = Result<graph::Graph, Error>> {
   | |______________________________________________________^
   |
   = note: `-D clippy::manual-async-fn` implied by `-D warnings`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_async_fn
help: make the function `async` and return the output of the future directly
   |
19 | async pub(crate) fn fetch_graph_from_gb(
20 |     upstream_base: reqwest::Url,
21 |     stream: String,
22 |     basearch: String,
23 |     req_timeout: Duration,
24 | ) -> Result<graph::Graph, Error> {
   |
help: move the body of the async block to the enclosing function
   |
24 | ) -> impl Future<Output = Result<graph::Graph, Error>> {
25 |     if stream.trim().is_empty() {
26 |         bail!("unexpected missing stream");
27 |     }
28 |     if basearch.trim().is_empty() {
29 |         bail!("unexpected missing basearch");

```

Reply:
I've updated the code to follow clippy's suggestion while at it, no real logic changes at play here.